modules/widgets/googleplus-badge.php

```diff
@@ -0,0 +1,309 @@
+					esc_attr( $instance['width'] ),
+					$instance['type']
+				);
+				break;
+			case 'community':
+				printf(
+					'<g:%s href="%s" layout="%s" theme="%s" showphoto="%s" showowners="%s" showtagline="%s" width="%s"></g:%s>',
+					$instance['type'],
+					esc_url( $instance['href'] ),
+					esc_attr( $instance['layout'] ),
+					esc_attr( $instance['theme'] ),
+					esc_attr( $instance['show_photo'] ? 'true' : 'false' ),
+					esc_attr( $instance['show_owners'] ? 'true' : 'false' ),
+					esc_attr( $instance['show_tagline'] ? 'true' : 'false' ),
+					esc_attr( $instance['width'] ),
+					$instance['type']
+				);
+				break;
+		}
+
+		echo $args['after_widget'];
+
+		do_action( 'jetpack_stats_extra', 'widget', 'googleplus-badge' );
+	}
+
+	function update( $new_instance, $old_instance ) {
+		$instance = array();
+
+		$instance['title'] = trim( strip_tags( stripslashes( $new_instance['title'] ) ) );
+
+		// Validate the Google+ URL
+		$instance['href'] = trim( strip_tags( stripslashes( $new_instance['href'] ) ) );
+
+		if ( $this->is_valid_googleplus_url( $instance['href'] ) ) {
+			$temp = explode( '?', $instance['href'] );
+			$instance['href'] = str_replace( array( 'http://plus.google.com', 'https://plus.google.com' ), 'https://plus.google.com', $temp[0] );
+		} else {
+			$instance['href'] = '';
+		}
+
+		$instance['theme']  = $this->filter_text( $new_instance['theme'],  $this->default_theme,  $this->allowed_themes );
+		$instance['layout'] = $this->filter_text( $new_instance['layout'], $this->default_layout, $this->allowed_layouts );
+
+		switch( $instance['layout'] ) {
+			case 'portrait':
+				$instance['width'] = $this->filter_int( (int) $new_instance['width'], $this->default_width, $this->max_width, $this->min_width_portrait );
+				break;
+			case 'landscape':
+				$instance['width'] = $this->filter_int( (int) $new_instance['width'], $this->default_width, $this->max_width, $this->min_width_landscape );
+				break;
+		}
+
+		if ( array_key_exists( $new_instance['type'], $this->allowed_types ) ) {
+			$instance['type'] = $new_instance['type'];
+		} else {
+			$instance['type'] = $this->default_type;
+		}
+
+		switch( $instance['type'] ) {
+			case 'person':
+			case 'page':
+				$instance['show_coverphoto'] = (bool) $new_instance['show_coverphoto'];
+				break;
+			case 'community':
+				$instance['show_photo']  = (bool) $new_instance['show_photo'];
+				$instance['show_owners'] = (bool) $new_instance['show_owners'];
+				break;
+		}
+
+		$instance['show_tagline'] = (bool) $new_instance['show_tagline'];
+
+		return $instance;
+	}
+
+	function form( $instance ) {
+		?>
+
+		<p>
+			<label for="<?php echo $this->get_field_id( 'title' ); ?>">
+				<?php _e( 'Title', 'jetpack' ); ?>
+				<input type="text" name="<?php echo $this->get_field_name( 'title' ); ?>" id="<?php echo $this->get_field_id( 'title' ); ?>" value="<?php echo esc_attr( $instance['title'] ); ?>" class="widefat" />
```

Comment:
We'll need to set defaults for all of these fields in wp-admin.  Seeing `undefined` notices for almost all fields when I first add the widget in wp-admin.